src/nix-build/nix-build.cc

```diff
@@ -235,7 +235,7 @@ static void main_nix_build(int argc, char * * argv)
         else if (*arg == "-")
             readStdin = true;
 
-        else if (*arg != "" && arg->at(0) == '-')
+        else if (!arg->empty() && (*arg)[0] == '-')
```

Comment:
Does that bring anything? A failing `at` call isn’t nice because it’s hard to track, but this is undefined if out-of-bounds which isn’t nicer

Reply:
ok, might not be worth it (it's argument parsing and can be really slow without impacting anything much...)

Reply:
also, this can't be out-of-bounds because C++ strings are always zero-terminated (`string_view`s aren't).